modules/plugins/languages/gleam.nix

```
@@ -0,0 +1,71 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}: let
+  inherit (builtins) attrNames;
+  inherit (lib.options) mkEnableOption mkOption;
+  inherit (lib.modules) mkIf mkMerge;
+  inherit (lib.lists) isList;
+  inherit (lib.types) enum either listOf package str;
+  inherit (lib.nvim.lua) expToLua;
+  inherit (lib.nvim.types) mkGrammarOption;
+
+  cfg = config.vim.languages.gleam;
+
+  defaultServer = "gleam";
+  servers = {
+    gleam = {
+      package = pkgs.gleam;
+      lspConfig = ''
+        lspconfig.gleam.setup{
+          capabilities = capabilities,
+          on_attach = default_on_attach,
+          cmd = ${
+          if isList cfg.lsp.package
+          then expToLua cfg.lsp.package
+          else ''{"${cfg.lsp.package}/bin/gleam", "lsp"}''
+        }
+        }
+      '';
+    };
+  };
+in {
+  options.vim.languages.gleam = {
+    enable = mkEnableOption "Gleam language support";
+
+    treesitter = {
+      enable = mkEnableOption "Gleam treesitter" // {default = config.vim.languages.enableTreesitter;};
+      package = mkGrammarOption pkgs "gleam";
+    };
+
+    lsp = {
+      enable = mkEnableOption "Gleam LSP support" // {default = config.vim.languages.enableLSP;};
+
+      server = mkOption {
+        type = enum (attrNames servers);
+        default = defaultServer;
+        description = "Gleam LSP server to use";
+      };
+
+      package = mkOption {
+        type = either package (listOf str);
+        default = servers.${cfg.lsp.server}.package;
+        description = "gleam LSP server package, or the command to run as a list of strings";
```

```suggestion
        description = "Gleam LSP server package, or the command to run as a list of strings";
```

It's capitalized everywhere else, but not here.